Source/KSCrash/Source/KSCrash/Recording/Tools/BSG_KSMachHeaders.h

```diff
@@ -0,0 +1,103 @@
+//
+//  BSG_KSMachHeaders.h
+//  Bugsnag
+//
+//  Created by Robin Macharg on 04/05/2020.
+//  Copyright © 2020 Bugsnag. All rights reserved.
+//
+
+#ifndef BSG_KSMachHeaders_h
+#define BSG_KSMachHeaders_h
+
+#import <mach/machine.h>
+
+/**
+ * An encapsulation of the Mach header - either 64 or 32 bit, along with some additional information required for
+ * detailing a crash report's binary images.
+ */
+typedef struct {
+    const struct mach_header *mh;      /* The mach_header - 32 or 64 bit */
+    uint64_t imageVmAddr;
+    uint64_t imageSize;
+    uint8_t *uuid;
+    const char* name;
+    cpu_type_t cputype;          /* cpu specifier */
+    cpu_subtype_t cpusubtype;    /* machine specifier */
+    intptr_t slide;
+} BSG_Mach_Binary_Image_Info;
+
+/**
+ * MARK: - A Dynamic array container
+ * See: https://stackoverflow.com/a/3536261/2431627
+ */
+typedef struct {
+    size_t used;
+    size_t size;
```

Comment:
Consider making these `uint32_t` to better match dyld and avoid a cast in `bsg_kscrw_i_writeBinaryImages`?